migrations/2024-08-17-144959_community-post-tags/up.sql

```diff
@@ -0,0 +1,18 @@
+-- a tag for a post, valid in a community. created by mods of a community
+CREATE TABLE community_post_tag (
+    id serial PRIMARY KEY,
+    ap_id text NOT NULL UNIQUE,
+    community_id int NOT NULL REFERENCES community (id) ON UPDATE CASCADE ON DELETE CASCADE,
+    name text NOT NULL,
+    published timestamptz NOT NULL DEFAULT now(),
+    updated timestamptz,
+    deleted timestamptz
+);
```

Comment:
How about renaming this to `post_tag` and later adding a flag `global: bool`? Then global tags wont need their separate table, query and api endpoints.

Reply:
yeah, if we only expect global tags then it makes sense I think. I'm just not sure whether we should plan for more unknowns, because

* the same tag elements are also incoming from other AP software, but those might have pretty different structure, so can potentially not be handled in the same table
* there are other types of tags we might want, for example foreign community tags (e.g. fact-checking posts in communities by third parties similar to Community Notes on twitter), or tags with values etc, that are also "post tags" but need to be handled differently as well

Reply:
> the same tag elements are also incoming from other AP software, but those might have pretty different structure, so can potentially not be handled in the same table

Everything that federates needs to fit into our data formats, otherwise it gets ignored completely.

> there are other types of tags we might want, for example foreign community tags (e.g. fact-checking posts in communities by third parties similar to Community Notes on twitter), or tags with values etc, that are also "post tags" but need to be handled differently as well

Sounds like an entirely separate feature so I wouldnt worry about it now.